www/caddy/src/opnsense/mvc/app/views/Pischem/Caddy/general.volt

```diff
@@ -0,0 +1,194 @@
+ # AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ # OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ # SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ # INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ # CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ # POSSIBILITY OF SUCH DAMAGE.
+ #}
+
+<script type="text/javascript">
+    $(document).ready(function() {
+        var data_get_map = {'frm_GeneralSettings':"/api/caddy/General/get"};
+        mapDataToFormUI(data_get_map).done(function(data){
+            // console.log("Fetched data:", data); // Log the fetched data
+            var generalSettings = data.frm_GeneralSettings.caddy.general;
+
+            // Populate TlsAutoHttps dropdown
+            var tlsAutoHttpsSelect = $('#caddy\\.general\\.TlsAutoHttps');
+            tlsAutoHttpsSelect.empty(); // Clear existing options
+            $.each(generalSettings.TlsAutoHttps, function(key, option) {
+                if (key !== "") {  // Filter out the unwanted "None" option
+                    tlsAutoHttpsSelect.append(new Option(option.value, key, false, option.selected === 1));
+                }
+            });
+
+            // Populate TlsDnsProvider dropdown
+            var tlsDnsProviderSelect = $('#caddy\\.general\\.TlsDnsProvider');
+            tlsDnsProviderSelect.empty(); // Clear existing options
+            $.each(generalSettings.TlsDnsProvider, function(key, option) {
+                if (key !== "") {  // Filter out the unwanted "None" option
+                    tlsDnsProviderSelect.append(new Option(option.value, key, false, option.selected === 1));
+                }
+            });
+
+            // Populate Trusted Proxies dropdown
+            var accesslistSelect = $('#caddy\\.general\\.accesslist');
+            accesslistSelect.empty(); // Clear existing options
+            $.each(generalSettings.accesslist, function(key, option) {
+                accesslistSelect.append(new Option(option.value, key, false, option.selected === 1));
+            });
+
+            // Refresh selectpicker for these dropdowns
+            $('.selectpicker').selectpicker('refresh');
+
+            // Function to show alerts in the HTML message area
+            function showAlert(message, type = "error") {
+                var alertClass = type === "error" ? "alert-danger" : "alert-success";
+                var messageArea = $("#messageArea");
+
+                // Stop any current animation, clear the queue, and immediately hide the element
+                messageArea.stop(true, true).hide();
+
+                // Now set the class and message
+                messageArea.removeClass("alert-success alert-danger").addClass(alertClass).html(message);
```

Comment:
Hi! sorry, have you tested this for XSS?

Reply:
Hey, could you clear me up on this matter? I thought everything I do is protected from XSS Attacks because the OPNsense puts a CSFR Token in the page when I'm authenticated.

If you found a vulnerability, please tell me how to mitigate it because I'm a little new at this. Thank you :)

Reply:
ah, sorry. false alarm ).
since you are using `.html()` this could potentially be XSS-vulnerable (https://jsfiddle.net/2TpHC/).
depending on the `message` value. but I missed that you `json_decode` validation message before returning.
so it should be safe. sorry )

Reply:
Hey you don't have to be sorry. I like that you looked at the code and help me out. Thank you ^^